spec/lib/appsignal/transaction_spec.rb

```diff
@@ -1568,6 +1568,43 @@ def to_s
       end
     end
 
+    context "when the error is already reported" do
+      it "does not add the error" do
+        # Report it on another transaction first
+        transaction1 = new_transaction
+        transaction1.add_error(error)
+        expect(transaction1).to have_error
+
+        transaction2 = new_transaction
+        transaction2.add_error(error)
+        expect(transaction2).to_not have_error
+      end
+    end
```

Comment:
Is this the desired behaviour? I understand that it is what's been implemented, but those being different transactions, I would expect reporting the same error to be reported both times.

Reply:
It's unlikely it happens at all, since we recommend using one transaction (that we duplicate to report multiple errors). And yes, my expectation would be that if we deduplicate errors in one transaction, we deduplicate them across all to avoid reporting the same error multiple times.

Reply:
Wouldn't this cause transaction two, which had an error added to it, to be reported as a successful transaction instead of an error transaction? Couldn't that mislead the customer about what is taking place?

Reply:
Perhaps?
Why are you trying to add the same error to two transactions?

Reply:
Hm, yeah, not sure how this would happen in practice.